subprojects/plugins/src/main/java/org/gradle/api/distribution/internal/DefaultDistribution.java

```diff
@@ -42,13 +42,13 @@ public String getName() {
 
     @Override
     public String getBaseName() {
-        DeprecationLogger.deprecateProperty("baseName").replaceWith("distributionBaseName").nagUser();
+        DeprecationLogger.deprecateProperty("baseName").replaceWith("distributionBaseName").withJavadoc(Distribution.class).nagUser();
```

Comment:
Same comments as above for AbstractArchiveTask.  I think we should incorporate the type into `deprecateProperty` and link specifically to the DSL reference for the type.  If we don't have a DSL reference, we can add it.